src/ft2font.cpp

```diff
@@ -176,15 +176,21 @@ inline double conv(long v)
 static FT_UInt ft_get_char_index_or_warn(FT_Face face, FT_ULong charcode)
 {
     FT_UInt glyph_index = FT_Get_Char_Index(face, charcode);
-    if (!glyph_index) {
-        PyErr_WarnFormat(NULL, 1, "Glyph %lu missing from current font.", charcode);
-        // Apparently PyErr_WarnFormat returns 0 even if the exception propagates
-        // due to running with -Werror, so check the error flag directly instead.
-        if (PyErr_Occurred()) {
-            throw py::exception();
-        }
+    if (glyph_index) {
+        return glyph_index;
+    }
+    PyObject *text_helpers = NULL, *tmp = NULL;
+    if (!(text_helpers = PyImport_ImportModule("matplotlib._text_helpers")) ||
+        !(tmp = PyObject_CallMethod(text_helpers, "warn_on_missing_glyph", "k", charcode))) {
+        goto exit;
```

Comment:
I don't really get the `goto exit`. It just jumps out of the `if` block. But the `exit:` label is the next statement anyway.

Reply:
This is a normal pattern for C extension functions (have an exit block at the end which just has all the XDECREFs and jump to it if anything fails).  See e.g. convert_open_args in ft2font_wrapper.  Here the function is simple enough that you *could* not use this pattern; I prefer keeping it but can rewrite the function without it if you insist.

Reply:
This will be more useful if more lines get added below, which is perhaps not very likely in this particular function, but who knows. In any case, I think the flow of execution is quite clear.